### PR TITLE
TextNAS without retrain

### PR DESCRIPTION
**Pull request by ultmaster**

### CODE REVIEW

File: examples/nas/textnas/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+python search.py
+```
+
+After each search epoch, 10 sampled architectures will be tested directly. Their performances are expected to be 40% - 42% after 10 epochs.
+
+By default, 20 sampled architectures will be exported into `checkpoints` directory for next step.
+
+## Retrain
+
+Not ready.
```
Comment (xuehui1991) [inline examples/nas/textnas/README.md]: ？

File: examples/nas/textnas/dataloader.py [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+        if line == '':
+            return None
+        # strip whitespace and only use if this contains something
+        line = line.strip()
+        if line == '':
+            continue
+        cur_text.append(line)
+        # Update depth
+        for char in line:
+            if char == '(':
```
Comment (xuehui1991) [inline examples/nas/textnas/dataloader.py]: can you unify the `'` and `"` in one file?

File: examples/nas/textnas/README.md (hunk 1 above)
Comment (ultmaster) [reply examples/nas/textnas/README.md]: We won't release retraining code in this release based on authors' comments.

File: examples/nas/textnas/search.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    device = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+    train_dataset, valid_dataset, test_dataset, embedding = read_data_sst("data")
+    train_loader = torch.utils.data.DataLoader(train_dataset, batch_size=args.batch_size, num_workers=4, shuffle=True)
+    valid_loader = torch.utils.data.DataLoader(valid_dataset, batch_size=args.batch_size, num_workers=4, shuffle=True)
+    test_loader = torch.utils.data.DataLoader(test_dataset, batch_size=args.batch_size, num_workers=4)
+    train_loader, valid_loader = cycle(train_loader), cycle(valid_loader)
+    model = Model(embedding)
+
+    num_epochs = 10
```
Comment (xuehui1991) [inline examples/nas/textnas/search.py]: make it as a parameter?

File: examples/nas/textnas/search.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    mutator = EnasMutator(model, temperature=None, tanh_constant=None, entropy_reduction="mean")
+
+    criterion = nn.CrossEntropyLoss()
+    optimizer = torch.optim.Adam(model.parameters(), lr=5e-3, eps=1e-3, weight_decay=2e-6)
```
Comment (xuehui1991) [inline examples/nas/textnas/search.py]: same here

File: examples/nas/textnas/search.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                             batch_size=args.batch_size,
+                             num_epochs=num_epochs,
+                             dataset_train=None,
+                             dataset_valid=None,
+                             train_loader=train_loader,
+                             valid_loader=valid_loader,
+                             test_loader=test_loader,
+                             log_frequency=args.log_frequency,
+                             mutator=mutator,
+                             mutator_lr=2e-3,
```
Comment (xuehui1991) [inline examples/nas/textnas/search.py]: same here

File: examples/nas/textnas/utils.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
+import logging
```
Comment (xuehui1991) [inline examples/nas/textnas/utils.py]: license?

File: src/sdk/pynni/nni/nas/pytorch/enas/mutator.py [hunk 7]
```diff
@@ -30,11 +30,12 @@ def forward(self, inputs, hidden):
 class EnasMutator(Mutator):
 
     def __init__(self, model, lstm_size=64, lstm_num_layers=1, tanh_constant=1.5, cell_exit_extra_step=False,
-                 skip_target=0.4, branch_bias=0.25):
+                 skip_target=0.4, temperature=None, branch_bias=0.25, entropy_reduction="sum"):
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/nas/pytorch/enas/mutator.py]: do you need to change the doc here? since the user will implement a customized EnasMutator

Comment (ultmaster) [reply src/sdk/pynni/nni/nas/pytorch/enas/mutator.py]: Let's do it separately.
